### PR TITLE
changed BookRepository for using JpaRepository

### PR DESCRIPTION
**Pull request by apanyutin**
implemented soft delete concept for the Book entity
added Liquibase support

### CODE REVIEW

File: src/main/java/com/example/controller/BookController.java [hunk 1]
```diff
@@ -5,15 +5,17 @@
...
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
-@RequestMapping("/books")
+@RequestMapping("/api/books")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

add `server.servlet.context-path=/api` to app properties

File: src/main/java/com/example/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -32,8 +31,25 @@ public List<BookDto> getAll() {
...
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(CreateBookRequestDto requestDto, Long id) {
+        Book book = bookRepository.getReferenceById(id);
+        book.setTitle(requestDto.getTitle());
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/service/impl/BookServiceImpl.java]: use mapstruct for this logic

File: src/main/java/com/example/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -32,8 +31,25 @@ public List<BookDto> getAll() {
...
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(CreateBookRequestDto requestDto, Long id) {
+        Book book = bookRepository.getReferenceById(id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/service/impl/BookServiceImpl.java]: https://www.baeldung.com/spring-data-jpa-getreferencebyid-findbyid-methods
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(()...
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**